hvac/api/auth/ldap.py

```diff
@@ -0,0 +1,367 @@
+        :param mount_point: The "path" the method/backend was mounted on.
+        :type mount_point: str | unicode
+        :return: The response of the configure request.
+        :rtype: requests.Response
+        """
+        params = {
+            'userdn': user_dn,
+            'groupdn': group_dn,
+            'url': url,
+            'case_sensitive_names': case_sensitive_names,
+            'starttls': starttls,
+            'tls_min_version': tls_min_version,
+            'tls_max_version': tls_max_version,
+            'insecure_tls': insecure_tls,
+            'certificate': certificate,
+            'userattr': user_attr,
+            'discoverdn': discover_dn,
+            'deny_null_bind': deny_null_bind,
+            'groupfilter': group_filter,
+            'groupattr': group_attr,
+        }
+        # Fill out params dictionary with any optional parameters provided
+        if upn_domain is not None:
+            params['upndomain'] = upn_domain
+        if bind_dn is not None:
+            params['binddn'] = bind_dn
+        if bind_pass is not None:
+            params['bindpass'] = bind_pass
+        if certificate is not None:
+            params['certificate'] = certificate
+
+        api_path = '/v1/auth/{mount_point}/config'.format(mount_point=mount_point)
+        return self._adapter.post(
+            url=api_path,
+            json=params,
+        )
+
+    def read_configuration(self, mount_point=DEFAULT_MOUNT_POINT):
+        """
+        Retrieve the LDAP configuration for the auth method.
+
+        Supported methods:
+            GET: /auth/{mount_point}/config. Produces: 200 application/json
+
+        :param mount_point: The "path" the method/backend was mounted on.
+        :type mount_point: str | unicode
+        :return: The JSON response of the read_configuration request.
+        :rtype: dict
+        """
+        api_path = '/v1/auth/{mount_point}/config'.format(mount_point=mount_point)
+        response = self._adapter.get(
+            url=api_path,
+        )
+        return response.json()
+
+    def create_or_update_group(self, name, policies=None, mount_point=DEFAULT_MOUNT_POINT):
+        """
+        Create or update LDAP group policies.
+
+        Supported methods:
+            POST: /auth/{mount_point}/groups/{name}. Produces: 204 (empty body)
+
+
+        :param name: The name of the LDAP group
+        :type name: str | unicode
+        :param policies: List of policies associated with the group. This parameter is transformed to a comma-delimited
+            string before being passed to Vault.
+        :type policies: list
+        :param mount_point: The "path" the method/backend was mounted on.
+        :type mount_point: str | unicode
+        :return: The response of the create_or_update_group request.
+        :rtype: requests.Response
+        """
+        if policies is None:
+            policies = []
+        params = {
+            'policies': ','.join(policies),
```

Comment:
I always ask myself in cases like this whether I want to enforce input types, or just say "rtfds" (read the frickin docstring).  this kind of thing has bitten me in the past:

```python
In [1]: policies = 'only-one-policy'

In [2]: ','.join(policies)
Out[2]: 'o,n,l,y,-,o,n,e,-,p,o,l,i,c,y'
```

If a bunk input type can result in Quite Bad Things, then I'll enforce the type (either with an exception, or help em out and make it a list), otherwise if the API (or whatever) handles it gracefully, just let the chips fall where they may.

Reply:
I added a ParamValidation exception a while back for this type of deal, would be reasonable to utilize it here. Could also make the case for casting things to a list if given a string where a list is expected, but I think its better to alert the user to the correct param type rather than hide it away implicitly...

Reply:
6210eeb. Touches on https://github.com/hvac/hvac/issues/129 to an extent.